src/NHibernate/Async/Multi/QueryBatchItemBase.cs

```diff
@@ -103,7 +103,7 @@ public async Task<int> ProcessResultsSetAsync(DbDataReader reader, CancellationT
 
 					queryInfo.Result = tmpResults;
 					if (queryInfo.CanPutToCache)
-						queryInfo.ResultToCache = tmpResults;
+						queryInfo.ResultToCache = new List<object>(tmpResults);
```

Comment:
When merging to master this change should be ignored. The following code is expected in master:
```C#
queryInfo.ResultToCache = queryCacheBuilder.Result;
```